### PR TITLE
Menu layout

### PR DESCRIPTION
**Pull request by ctlai95**
Screen Shot 2020-05-30 at 2 13 02 PM

Here's my attempt at a two-menu layout with a scrollable section to select the food category. Images are placeholders for now. Not mobile friendly yet.

### CODE REVIEW

**Comment by AlvinDevelopments** [comment]
Do you think the banner "X" would look better if aligned with the right caret arrow?

**Comment by ctlai95** [comment]
@AlvinDevelopments you mean moving it away from the "Free Delivery" text?

File: src/components/Banner/Banner.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+const Banner = (props: IProps) => {
+  const [dismissed, setDismissed] = useState(false);
+  const [tooltipHidden, setTooltipHidden] = useState(true);
+
+  return (
+    <div className={`banner ${dismissed && 'dismissed'}`}>
+      <div
+        onMouseEnter={() => setTooltipHidden(false)}
+        onMouseLeave={() => setTooltipHidden(true)}
```
Comment (AlvinDevelopments) [inline src/components/Banner/Banner.tsx:19]: Would a `toggleTooltip` function work better here? As that will only require creating a single handler function to eliminate the need for anon functions

File: src/components/Banner/Banner.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+      >
+        <FontAwesomeIcon icon={faInfoCircle} />
+        <div className={`tooltip ${tooltipHidden && 'hidden'}`}>
+          {props.tooltipMessages.map((message, index) => (
```
Comment (AlvinDevelopments) [inline src/components/Banner/Banner.tsx:23]: Worth to destructure `props`?

File: src/components/Menu/Menu.tsx [hunk 3]
```diff
@@ -1,6 +1,94 @@
...
+    <div className="menu">
+      <Banner message={menu.delivery} tooltipMessages={menu.discounts} />
+      <div className="slider-container">
+        <div className="left" onClick={() => scroll(-500)}>
+          <FontAwesomeIcon icon={faChevronLeft} size="lg" />
+        </div>
+        <div ref={sliderRef} className="slider">
+          {menu.categories.map((category: ICategory, index) => (
+            <div
+              key={index}
```
Comment (AlvinDevelopments) [inline src/components/Menu/Menu.tsx:58]: I would suggest using something other than `index` as the key, as it's not reliable. Maybe `category.name[language]`?

File: src/components/Menu/Menu.tsx [hunk 4]
```diff
@@ -1,6 +1,94 @@
...
+      </div>
+      <div className="active-category">
+        <h1>{activeCategory.name[language]}</h1>
+        <table>
+          <tbody>
+            {activeCategory.items.map((item: IItem, index) => (
+              <tr key={index}>
+                <td className="number">{item.number}</td>
+                <td className="name">{item.name[language]}</td>
+                <td className="price">{item.price}</td>
```
Comment (AlvinDevelopments) [inline src/components/Menu/Menu.tsx:84]: I would suggest destructuring here so you don't have to repeat `item` :)

**Comment by ctlai95** [comment]
Thanks @AlvinDevelopments, all good suggestions. I'll make those changes

**Comment by AlvinDevelopments** [comment]
Screen Shot 2020-05-30 at 2 47 16 PM

Oh I was thinking maybe aligning the close button with the prices would make it "flow" better

File: src/components/Menu/Menu.tsx (hunk 3 above)
Comment (ctlai95) [reply src/components/Menu/Menu.tsx:58]: After reading up on keys a bit I would argue that `index` as key makes sense for static lists (nothing changes unless `menu.json` changes). Thoughts?

**Comment by ctlai95** [comment]
@AlvinDevelopments I actually prefer the "X" closer for symmetry, but that's just me. I suppose the styling would eventually need to change though when we think about responsive design for mobile devices

File: src/components/Menu/Menu.tsx (hunk 3 above)
Comment (AlvinDevelopments) [reply src/components/Menu/Menu.tsx]: Yea that's fine then as `category.name[language]` could be clunky. If there were no issues with this during testing it should be fine to use

**Pull request merged**